src/main.rs

```diff
@@ -0,0 +1,188 @@
+use std::env;
+use std::fmt;
+use std::fs::{self,File};
+use std::io::{self,Read};
+use std::os::unix::fs as unix_fs;
+use std::path::{Path,PathBuf};
+use std::process::Command;
+
+struct PathError {
+    message: String,
+    path: PathBuf,
+}
+
+impl fmt::Display for PathError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}: {}", self.message, self.path.display())
+    }
+}
+
+// Creates a file if it does not exist
+fn ensure_file_exists<P: AsRef<Path>>(file_path: P, is_directory: Option<bool>) -> io::Result<()> {
+    let path = file_path.as_ref();
+
+    if is_directory == Some(true) {
+        if !path.exists() {
+            fs::create_dir_all(path)?;
+        };
+    } else {
+        if !path.exists() {
+            File::create(path)?;
+        };
+    }
+
+    Ok(())
+}
+
+// Create config files if they don't exist
+fn ensure_config_files_exist(base16_config_path: &Path, base16_shell_theme_name_path: &Path) -> Result<(), PathError> {
+    ensure_file_exists(base16_config_path, Some(true)).map_err(|_| PathError {
+        message: "Failed to create directory. Check if parent directory exists.".to_string(),
+        path: base16_config_path.to_path_buf()
+    })?;
+    ensure_file_exists(base16_shell_theme_name_path, Some(false)).map_err(|_| PathError {
+        message: "Failed to create directory. Check if parent directory exists.".to_string(),
```

Comment:
I think this message is wrong, since this is using ensure_file_exists with is_directory set to false.